src/Services/SocialUserLoginService.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Blumilk\Meetup\Core\Services;
+
+use Blumilk\Meetup\Core\Models\User;
+use Illuminate\Auth\AuthManager;
+use Illuminate\Contracts\Hashing\Hasher;
+use Illuminate\Session\Store;
+
+class SocialUserLoginService
+{
+    public function __construct(
+        public AuthManager $authManager,
+        public Hasher $hasher,
+        public Store $session,
+    ) {}
+
+    public function registerOrLogin($socialUser, string $provider): void
```

Comment:
Please type your `$socialUser` parameter.

Reply:
Or maybe passing email and name would be more verbose than using some anonymous array?